DuckDB.NET.Data/DuckDBDataReader.cs

```diff
@@ -191,17 +191,28 @@ public override string GetString(int ordinal)
 
     public override T GetFieldValue<T>(int ordinal)
     {
-        return vectorReaders[ordinal].ColumnDuckDBType switch
+        var value = vectorReaders[ordinal].ColumnDuckDBType switch
         {
-            DuckDBType.List => (T)vectorReaders[ordinal].GetList(rowsReadFromCurrentChunk - 1, (dynamic)Activator.CreateInstance<T>()),
-            DuckDBType.Enum => vectorReaders[ordinal].GetEnum<T>(rowsReadFromCurrentChunk - 1),
-            _ => (T)vectorReaders[ordinal].GetValue(rowsReadFromCurrentChunk - 1)
+            DuckDBType.List => vectorReaders[ordinal].GetList(rowsReadFromCurrentChunk - 1, Activator.CreateInstance<T?>() as dynamic),
+            DuckDBType.Enum => vectorReaders[ordinal].GetEnum<T>(rowsReadFromCurrentChunk - 1) ,
+            _ => vectorReaders[ordinal].GetValue(rowsReadFromCurrentChunk - 1)
         };
+
+        if (value is not null)
+            return (T)value;
+
+        if (default(T) is null && typeof(T).IsValueType)
+            return default!;
+
+        if (typeof(T) == typeof(object))
```

Comment:
I'm not sure if this method should ever return DBNull

Reply:
The method returns T and we already know that it's not a Nullable<T> ... returning `DBNull.Value` let's us return an object.